habit/src/AppScreen.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { createAppContainer } from 'react-navigation';
+
+import { createRootSwitchNavigator } from './AppNavigation';
+import SplashScreen from './containers/SplashScreen';
+
+export default class AppScreen extends React.Component {
+    state = {
+        isLoading: true,
+        isSignedOut: true,
+        userToken: null,
+        isTimeDone: false,
+    };
+
+    componentDidMount() {
+        // eslint-disable-next-line consistent-this
+        let self = this;
+        this.checkCredentialsAndUpdate();
+        // NOTE: This is to forcefully make the SplashScreen last for 2 minutes
+        setTimeout(function() {
+            self.setState({ isTimeDone: true });
+        }, 2000);
+    }
+
+    async checkCredentialsAndUpdate() {
```

Comment:
why is this function async, when we are not using it as an asynchronous function?

Reply:
This function will be async as soon as we implement the login API.
We will have to store the login credentials in a 3rd party library like Keychain (get from such libraries is async)